lib/collection/request.js

```diff
@@ -107,6 +107,38 @@ _.assign(Request.prototype, /** @lends Request.prototype */ {
     },
 
     /**
+     * Sets authentication method for the request
+     *
+     * @param {?String|RequestAuth~definition} type
+     * @param {VariableList~definition=} [options]
+     *
+     * @note This function was previously (in v2 of SDK) used to clone request and populate headers. Now it is used to
+     * only set auth information to request
+     *
+     * @note that ItemGroup#authorizeUsing depends on this function
+     */
+    authorizeUsing: function (type, options) {
+        if (_.isObject(type) && _.isNil(options)) {
```

Comment:
@shamasis should this be `_.isObject(type) && !_.isNil(options)`

Reply:
Done